Add OPL tutorial (#423)
oasisprotocol/docs: docs/dapp/opl/README.md

> # OPL

Reviewer: ```suggestion
# Oasis Privacy Layer
```

> The Oasis Privacy Layer (OPL) is an EVM-compatible privacy solution that opens

Reviewer: ```suggestion
The Oasis privacy layer (OPL) is an EVM-compatible privacy solution that opens
```

new frontiers for confidential smart contracts in Web3 applications on the
most popular EVM networks like Ethereum, BNB Chain, and Polygon.

## DApp Development

The Oasis Privacy Layer contains a set of [smart contracts](https://github.com/oasisprotocol/sapphire-paratime/tree/main/contracts/contracts/opl)

> for [Sapphire](../sapphire/README.mdx) to empower your existing and future Web3

Reviewer: ```suggestion
that integrate [Oasis Sapphire](../sapphire/README.mdx) into your existing and future Web3
```

> on the [Sapphire](../sapphire/README.mdx) paratime layer to empower your existing and future Web3

Reviewer: Perhaps just:

```suggestion
for [Sapphire](../sapphire/README.mdx) to empower your existing and future Web3
```

> on the [Sapphire](../sapphire/README.mdx) paratime layer to empower your existing and future Web3

Reviewer: Perhaps just:

```suggestion
for [Sapphire](../sapphire/README.mdx) to empower your existing and future Web3
```

> applications with confidentiality.

Reviewer: ```suggestion
applications and introduce confidentiality.
```
